xarray/coding/variables.py

```diff
@@ -537,6 +537,7 @@ def _choose_float_dtype(
     if dtype.itemsize <= 2 and np.issubdtype(dtype, np.integer):
         return np.float32
     # For all other types and circumstances, we just use float64.
+    # Todo: with nc-complex from netcdf4-python >= 1.7.0 this is available
```

Comment:
Note to self: Add a test for this and fix it!

Reply:
Maybe leave it to dedicated follow-up PR?

Reply:
Will this be an issue? nc-complex still stores the data as floats, just either in a compound type or with an extra dimension (netCDF-3).

Although, HDF5 1.15 will have native support for complex numbers, so that may be a future consideration (I suspect it will still be some time before netCDF itself gets native support)

Reply:
Yes, maybe it will work out of the box.